Library/Homebrew/formulary.rb

```diff
@@ -237,18 +245,74 @@ def self.loader_for(ref)
     return NullLoader.new(ref)
   end
 
+  def self.find_with_priority(ref, behavior)
+    available_formulas = tap_paths(ref)
+    core_formula = core_path(ref)
+    if core_formula.file?
+      available_formulas[Tap.core_priority] = [] if available_formulas[Tap.core_priority].nil?
+      available_formulas[Tap.core_priority] << core_formula
+    end
+    unless available_formulas.empty?
+      if behavior == FactoryBehavior::ENFORCE_UNIQUE
+        flat_formulas = available_formulas.values.flatten
+        if flat_formulas.length == 1
+          return flat_formulas.first
+        else
+          raise TapFormulaAmbiguityError.new(ref, flat_formulas)
+        end
+      else
+        if available_formulas.length > 1
+          opoo "Some formulae are excluded due to priority settings. Run brew info #{ref} to see them."
+        end
+        available_formulas.keys.sort.each do |this_priority|
+          if available_formulas[this_priority].length > 1
+            if behavior == FactoryBehavior::INSTALL_LIKE
+              ohai "Multiple formulae with same name and priority available."
+              formulae_list = available_formulas[this_priority]
+              puts_columns (0..formulae_list.length-1).map { |i| "#{i+1}. #{formulae_list[i].fully_qualified_name}"}
+              print "Please choose one by inputing the number before it: "
+              selected_index = $stdin.readline.to_i - 1
```

Comment:
The typical Homebrew way of solving stuff like this is to error-out and demand the user rerun with an more information e.g. fully specify the tap with `homebrew/versions/gcc48`